### PR TITLE
Fixed using languages with country codes

### LINKED ISSUE #5451: Selecting languages that contain country codes does not work
#### Problem description
Selecting languages that contain country codes does not work

#### Steps to reproduce the problem
Install the app and try to set the language to Swedish, Lao, Kiswahili (Kenya), or Nepali.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5451 

#### What has been done to verify that this works as intended?
I've verified the fix manually.

#### Why is this the best possible solution? Were any other approaches considered?
The problem was that in the case of languages that contain both language and country code, we need to use a proper constructor to create a `Locale` object see https://github.com/getodk/collect/pull/5452/files#diff-b60064e0aecff97e0fc520565d2533c6ccf51822367bcefcf0f3a1ae42c87217R33.
In https://github.com/getodk/collect/pull/5452/files#diff-a2f0be8cb9ce6618f33c478f6ae0c69fc5e6eefa5807833ffc3f1f324369c5b3L261 we didn't do that and that was the problem. Now we use the `LocaleHelper` class that does that properly.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Please test using other languages too not only those I mentioned in the issue.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/MainMenuPage.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public class MainMenuPage extends Page<MainMenuPage> {
     @Override
     public MainMenuPage assertOnPage() {
         return WaitFor.waitFor(() -> {
-            onView(withText(R.string.enter_data)).check(matches(isDisplayed()));
+            onView(withText(R.string.enter_data_button)).check(matches(isDisplayed()));
```
Comment (grzesiek2010) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/pages/MainMenuPage.java:30]: It was a bug.
`enter_data` is a title displayed when we open the list of forms
`enter_data_button` is the title of the button we use to open the list of forms.
In English, both hold the same string but it's not always the case in all other languages.

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/UserInterfacePage.java [hunk 2]
```diff
@@ -22,8 +26,15 @@ public UserInterfacePage clickOnLanguage() {
     }
 
     public MainMenuPage clickOnSelectedLanguage(String language) {
-        onView(withText(language)).perform(ViewActions.scrollTo());
-        onView(withText(language)).perform(click());
+        try {
+            onView(withText(language)).perform(click());
+        } catch (NoMatchingViewException e) {
+            for (int i = 0; i < 10; i++) {
```
Comment (grzesiek2010) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/pages/UserInterfacePage.java:32]: Scrolling didin't work in that list of languages so I used this nasty workaround.

**Review by lognaturel — APPROVED** [review]

**Comment by dbemke** [comment]
If a user sets language to Kiswahili (Kenya) there is no subtext (added/updated etc.) under the name of a form e.g in Fill Blank Form or View Sent Form. There is only the version of a form. 
kiwassubtext

**Comment by dbemke** [comment]
I checked all languages and some aren't translated:
- lingala - I saw there is a cancel button translated (so maybe it is translated)
- Malagasy 
- lg1

- lg2

**Comment by grzesiek2010** [comment]
Lingala works well but it has only a few translations so it's difficult to notice that. One of them is the first button in Image widgets:
Screenshot_1678186744

Malagasy the same as above:
Screenshot_1678186893

Tigrinya is not translated at all.

Sinhala also only a few translations:
Screenshot_1678187158

So everything looks fine to me.

**Comment by grzesiek2010** [comment]
> If a user sets language to Kiswahili (Kenya) there is no subtext (added/updated etc.) under the name of a form e.g in Fill Blank Form or View Sent Form. There is only the version of a form.

This is not related, please file a separate issue.

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on Android 10, 13

**Verified Cases:**
- issue #5451 is no longer reproducing
- all languages available in settings

The issue mentioned in the PR will be filled separately.

**Pull request merged**